### PR TITLE
Getting the python client into a reasonable state

### PR DESCRIPTION
**Pull request by SandersAaronD**
Presently if all other services work (big if) we should be able to do the following:

```bash
export GF_AITRAINING_CREDS=83bcaff6228b39bbe431af5e19fb4368e2a03dd3@localhost:3456
```

where port number is whatever port number we expose the API at in the dev environment

We can then do:

```python
import o11y
import random  # for demo script

epochs = 10
lr = 0.01

run = o11y.init(
    # Set the project where this run will be logged
    project="my-awesome-project",
    # Track hyperparameters and run metadata
    config={
        "learning_rate": lr,
        "epochs": epochs,
    },
)

offset = random.random() / 5
print(f"lr: {lr}")

# simulating a training run
for epoch in range(2, epochs):
    acc = 1 - 2**-epoch - random.random() / epoch - offset
    loss = 2**-epoch + random.random() / epoch + offset
    print(f"epoch={epoch}, accuracy={acc}, loss={loss}")
    o11y.log({"accuracy": acc, "loss": loss})
```

For getting other services to where we can test this end to end,
- it's totally fine to ignore the "project" and "run" inputs from the call to register_process initially
- We need register_process and the proxy to loki for custom logs to work

Only other serious todo for logging is to set up redirecting stdout to logs; this is actually harder than it appears, and I'll try to write up a separate issue for it.

Metrics should be TBD for best approach (beyla etc) because when we did this previously in the prototype it quickly devolved into a lot of custom ad hoc code for each individual metric we wanted to expose, and it would be much nicer to use existing instrumentation (which we are also likely to need anyway for e.g. gpu logs).

### CODE REVIEW

File: o11y/o11y/_internal/client.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import logging_loki
+import logging
+from .. import logger
+
+
+class Client:
+    def __init__(self):
+        # We are going to assume that the user has set the credentials in the environment
+        # There are other flows but it's the easiest one
+        login_string = os.environ.get('GF_AI_TRAINING_CREDS')
```
Comment (annanay25) [inline o11y/o11y/_internal/client.py:18]: nit: should we name this `GF_AI_TRAINING_TOKEN`?

**Review by annanay25 — APPROVED** [review]
LGTM, bar a minor nit

File: o11y/o11y/_internal/client.py (hunk 1 above)
Comment (SandersAaronD) [reply o11y/o11y/_internal/client.py:18]: I have it as "creds" because it's got a URL in it too at the moment; because "grafana" isn't one specific URL and users each have their own environment we can't really have a "default", so to preserve minimal setup steps we have one environment variable that contains @-separated token + url.
